File: backend/SindhDormitory.Domain/Entities/HostelImage.cs
// =============================================================================
// Domain/Entities/HostelImage.cs
// =============================================================================
namespace SindhDormitory.Domain.Entities;

public class HostelImage : BaseEntity
{
    public int    ImageId    { get; set; }
    public int    HostelId   { get; set; }
    public string ImageUrl   { get; set; } = string.Empty;
    public bool   IsPrimary  { get; set; } = false;
    public string? Caption   { get; set; }

    // Navigation
    public Hostel Hostel { get; set; } = null!;
}

File: backend-node/src/entities/HostelImage.ts
import {
  Entity, PrimaryGeneratedColumn, Column, ManyToOne, JoinColumn,
  CreateDateColumn, UpdateDateColumn
} from 'typeorm';
import { Hostel } from './Hostel';

@Entity('HostelImages')
export class HostelImage {
  @PrimaryGeneratedColumn({ name: 'ImageId' })
  imageId!: number;

  @Column({ name: 'HostelId', type: 'int' })
  hostelId!: number;

  @Column({ name: 'ImageUrl', type: 'varchar', length: 500 })
  imageUrl!: string;

  @Column({ name: 'IsPrimary', type: 'boolean', default: false })
  isPrimary!: boolean;

  @CreateDateColumn({ name: 'CreatedAt' })
  createdAt!: Date;

  @UpdateDateColumn({ name: 'UpdatedAt' })
  updatedAt!: Date;

  @ManyToOne(() => Hostel, (h) => h.images, { onDelete: 'CASCADE' })
  @JoinColumn({ name: 'HostelId' })
  hostel!: Hostel;
}
